src/test/java/com/marceldias/resource/MastermindResourceTests.java

```diff
@@ -0,0 +1,215 @@
+package com.marceldias.resource;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.jayway.restassured.RestAssured;
+import com.jayway.restassured.http.ContentType;
+import com.jayway.restassured.internal.mapping.Jackson2Mapper;
+import com.jayway.restassured.mapper.ObjectMapper;
+import com.jayway.restassured.mapper.factory.Jackson2ObjectMapperFactory;
+import com.marceldias.MastermindApplication;
+import com.marceldias.json.factory.GameJsonFactory;
+import com.marceldias.model.*;
+import com.marceldias.repository.GameRepository;
+import com.marceldias.service.GameService;
+import com.marceldias.service.GuessService;
+import org.apache.http.HttpStatus;
+import org.hamcrest.Matchers;
+import org.hamcrest.collection.IsArray;
+import org.hamcrest.collection.IsCollectionWithSize;
+import org.hamcrest.collection.IsEmptyCollection;
+import org.hamcrest.core.*;
+import org.hamcrest.text.IsEmptyString;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.IntegrationTest;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.web.WebAppConfiguration;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = MastermindApplication.class)
+@WebAppConfiguration
+@IntegrationTest("server.port:0")
+public class MastermindResourceTests {
+
+    @Autowired
+    private GameRepository gameRepository;
+
+    @Autowired
+    private GameService gameService;
+
+    @Autowired
+    private GuessService guessService;
+
+    @Value("${local.server.port}")
+    private int port;
+
+    private Game game;
+
+    private User user;
+
+    @Before
+    public void setUp() {
+
+        user = new User("test");
+        game = new Game(user);
+
+        gameRepository.deleteAll();
+        gameRepository.save(game);
+
+        RestAssured.port = port;
```

Comment:
\[FindBugs\] WARNING: ST: Write to static field com\.jayway\.restassured\.RestAssured\.port from instance method com\.marceldias\.resource\.MastermindResourceTests\.setUp\(\)